src/utils/deprecated/deprecated.tsx

```diff
@@ -5,14 +5,34 @@
 
 import React, { useEffect } from 'react';
 
+interface IDeprecatedComponentWarning {
+  NewComponent: React.ComponentType<any>;
+  version: string;
+}
+
+interface IDeprecatedPropWarning {
+  deprecatedProp: any;
+  deprecatedPropName: string;
+  version: string;
+}
+
 export const getDeprecatedMessage = (message: string): string =>
   `[DEPRECATED] ${message}`;
 
-export const deprecated = (message: string) => {
+export const deprecatedComponentWarning = ({
```

Comment:
Have we shipped a version with `deprecated` already? If so, that name should stay and be deprecated and alias to this function instead of plainly renaming it